src/dom/position/matrix.ts

```diff
@@ -0,0 +1,182 @@
+/**
+ * Class for dealing with CSS Transform matrices.
+ * Supports relative updates being made to a matrix from multiple pieces of
+ * code without them over-riding each other (unless explicitly desired).
+ */
+import { Numeric } from '../../types';
+import { Vector2d } from '../../mathf/geometry/vector-2d';
+import { ArrayMap } from '../../map/array';
+import { ComputedStyleService } from '../computed-style-service';
+import { Raf } from '../..';
+
+// Tracks desired changes to element matrices for the next frame
+const matrixChangesByElement: ArrayMap<HTMLElement, Matrix> = new ArrayMap();
+// Tracks starting state of each matrix for the current frame
+const preChangeMatrixByElement: Map<HTMLElement, Matrix> = new Map();
+
+// For figuring out how much to trim off strings to get at the values
+const STYLE_STRING_PREFIX = 'matrix(';
+const STYLE_STRING_PREFIX_LENGTH = STYLE_STRING_PREFIX.length;
+
+// Optimization to minimize DOM queries where possible
+const computedStyleService = ComputedStyleService.getSingleton();
+
+export class Matrix {
```

Comment:
My gut tells me maybe we should call this DomMatrix since it's specialized to that purpose.

Reply:
Done.

Used MatrixDom to following existing convention from VectorDom